aws_grant_user_access/src/process_event.py

```diff
@@ -70,3 +73,44 @@ def publish_sns_message(message: SNSMessage) -> None:
         SNSMessagePublisher(config.get_sns_client()).publish_sns_message(
             sns_topic_arn=sns_topic_arn, message=json.dumps(message.to_dict())
         )
+
+
+def _validate_role(role_arn: str) -> Any:
+    PERMITTED_ROLES = ["engineer"]
```

Comment:
Both for this and the user groups it feels a bit brittle that we're doing a string comparison on the ARN. 

- If someone adds a new role/group that they expect to be able to assume they have to know to update two places (aws_users and here) which is probably more coupled than we'd like
- If someone is trying to give a role this ability when it shouldn't have it, being able to just put "engineer" in the role name seems like something that's easy to not know or spot the implications of for a reviewer. 

Ideally this would be more intrinsically linked to a property of the role or something but I don't have a good suggestion for how to do that right now so will just leave this here for posterity.